### PR TITLE
IT-12787 Fix MultiStore Bug

### PR DESCRIPTION
**Pull request by SyedAbdullahShah**
Fix MultiStore Bug in Configuration
Ticket: [IT-12787](https://ekomi01.atlassian.net/browse/IT-12787

### CODE REVIEW

File: Model/Validate.php [hunk 1]
```diff
@@ -87,9 +95,19 @@ public function beforeSave()
...
+            $shopId = $postValues['shop_id']['value'];
+        } elseif (isset($postValues['shop_id']['inherit']) && $postValues['shop_id']['inherit'] == 1) {
+            $shopId = $this->dataHelper->getShopId($storeId);
+        }
+
+        if (isset($postValues['shop_password']['value'])) {
+            $shopPassword = $postValues['shop_password']['value'];
+        } elseif (isset($postValues['shop_password']['inherit']) && $postValues['shop_password']['inherit'] == 1) {
+            $shopPassword = $this->dataHelper->getShopPw($storeId);
+        }
```
Comment (amanmukhtar) [inline Model/Validate.php]: $shopId and $shopPassword are not defined for all paths

File: Model/Validate.php [hunk 2]
```diff
@@ -19,6 +19,7 @@
 use Magento\Framework\HTTP\Client\Curl;
 use Magento\Framework\Phrase;
 use Magento\Framework\Exception\LocalizedException;
+use Ekomi\EkomiIntegration\Helper\Data as DataHelper;
```
Comment (amanmukhtar) [inline Model/Validate.php]: Try not to keep things general. Here DataHelper is a general name for alias. You can name it like EkomiIntegrationHelper, etc or something like that which somewhat indicates the helper class purpose

File: Model/Validate.php [hunk 3]
```diff
@@ -103,6 +121,11 @@ public function beforeSave()
                 $this->activateSrrCustomerSegment($shopId, $shopPassword, $customerSegment["id"]);
             }
 
+            if (isset($postValues['smart_check']['value'])) {
+                $smartCheck = $postValues['smart_check']['value'];
+            } elseif (isset($postValues['smart_check']['inherit']) && $postValues['smart_check']['inherit'] == 1) {
+                $smartCheck = $this->dataHelper->getShopPw($storeId);
```
Comment (amanmukhtar) [inline Model/Validate.php]: Are you sure you want to store shop password in $smartCheck?

File: Model/Validate.php [hunk 4]
```diff
@@ -87,9 +95,19 @@ public function beforeSave()
...
+            $shopId = $postValues['shop_id']['value'];
+        } elseif (isset($postValues['shop_id']['inherit']) && $postValues['shop_id']['inherit'] == 1) {
+            $shopId = $this->dataHelper->getShopId($storeId);
+        }
+
+        if (isset($postValues['shop_password']['value'])) {
+            $shopPassword = $postValues['shop_password']['value'];
+        } elseif (isset($postValues['shop_password']['inherit']) && $postValues['shop_password']['inherit'] == 1) {
+            $shopPassword = $this->dataHelper->getShopPw($storeId);
+        }
```
Comment (amanmukhtar) [inline Model/Validate.php]: The above code is somewhat duplicate. You can make the above code a function and may be pass an index to check
```
$valueToReturn = '';
if (isset($postValues[$index]['value'])) {
    $valueToReturn = $postValues[$index]['value'];
} elseif (isset($postValues[$index]['inherit']) && $postValues[$index]['inherit'] == 1) {
    $valueToReturn = $this->getValue($index, $storeId); //here getValue can call the functions on the basis of index given
}
```
Check if something like above is appropriate for u

File: Model/Validate.php (hunk 2 above)
Comment (ghulammurtaza) [reply Model/Validate.php]: datahelper is default helper of module, magento's convention.

Comment (amanmukhtar) [reply Model/Validate.php]: Please ignore this point then. thanks for the clarity :)

File: Model/Validate.php (hunk 3 above)
Comment (SyedAbdullahShah) [reply Model/Validate.php]: fixed.

**Pull request merged**